### ISSUE TITLE
Specifying Loss Distributions for Temporal Fusion Transform

### ISSUE THREAD

**Issue by esbraun** [awslabs/gluon-ts#2168]
## Description
Currently the mxnet temporal fusion transformer only allows for quantile loss. It would be helpful to be able to specify other distributions (i.e. negative binomial for count data).

Would this change be possible?

## References
The library pytorch_forecasting allows specification of the loss for its [temporal fusion transformer implementation.](https://pytorch-forecasting.readthedocs.io/en/stable/api/pytorch_forecasting.models.temporal_fusion_transformer.TemporalFusionTransformer.html)

**Comment by kashif** [comment]
note I have such a version for pytorch: https://github.com/kashif/pytorch-transformer-ts/tree/main/tft

**Comment by esbraun** [comment]
Thank you! Is this something that you're planning on merging eventually as well?

**Comment by kashif** [comment]
yes eventually

**Comment by esbraun** [comment]
Update: I've empirically found significant improvement using the negative binomial distribution for a operations related count valued forecasting problem using the TFT implementation from kashif's repo over the mxnet TFT gluonts implementation and other gluonts algorithms. The implementation, however, lacks the full featured covariate support of the gluonts mxnet model. Hopefully the support for more loss functions could be wrapped into remediation for #2416.
